pkg/capture/capture.go

```diff
@@ -352,7 +408,7 @@ func newCaptureLock() *captureLock {
 	return &captureLock{
 		request: make(chan struct{}, 1),
 		confirm: make(chan struct{}),
-		done:    make(chan struct{}),
+		done:    make(chan struct{}, 1),
```

Comment:
Not only can it be buffered now, it has to be (unless I want to use the slower `select` approach). The reason is that I need to check for len() > 0 (in order to decide whether to consume without actually trying / `select`ing. Since this is only called from the `capture()` routine to unlock (and you can't enter the lock twice) it's safe + non-racy.